dataflow/gpu-workers/pytorch-minimal/e2e_test.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+try:
+    # `conftest` cannot be imported when running in `nox`, but we still
+    # try to import it for the autocomplete when writing the tests.
+    from conftest import Utils
+except ModuleNotFoundError:
+    Utils = None
+import pytest
+
+NAME = "dataflow/gpu-workers/pytorch-minimal"
+
+
+@pytest.fixture(scope="session")
+def bucket_name(utils: Utils) -> str:
+    yield from utils.storage_bucket(NAME)
+
+
+@pytest.fixture(scope="session")
+def build_image(utils: Utils) -> str:
+    yield from utils.cloud_build_submit(
+        image_name=NAME,
+        config="build.yaml",
+        substitutions={"_IMAGE": f"{NAME}:{utils.uuid}"},
+    )
+
+
+@pytest.fixture(scope="session")
```

Comment:
Does this mean you only run one dataflow job per session? It doesn't seem to make sense as the test suite should run as many jobs as requested, on the other hand it's fine to create only one bucket for test session if you'd like to reuse the same bucket.

Reply:
It's one job for this specific test suite. It doesn't really make any difference since there is only one test that uses this.

In case we wanted to check different things about the job, it would make sense to only launch one job and just poll different properties from it, although in this case we're only waiting until it finishes successfully.

Reply:
Also, if we sent multiple jobs, they would have the same job name so subsequent jobs would fail. For these tests, having exactly one job is what we want. It could be different in other test suites, but each one defines how they run their jobs.